lib/mix/tasks/docs.ex

```diff
@@ -113,25 +113,35 @@ defmodule Mix.Tasks.Docs do
   def run(args, config \\ Mix.Project.config, generator \\ &ExDoc.generate_docs/3) do
     Mix.Task.run "compile"
 
-    {cli_opts, args, _} = OptionParser.parse(args,
-                            aliases: [n: :canonical, f: :formatter, o: :output],
-                            switches: [canonical: :string, formatter: :keep, language: :string, output: :string])
-
-    if args != [] do
-      Mix.raise "Extraneous arguments on the command line"
+    {cli_options, remaining_args, _invalid} =
+      OptionParser.parse(args,
+        aliases: [n: :canonical, f: :formatter, o: :output],
+        switches: [formatter: :keep]
+      )
+
+    if remaining_args != [] do
+      Mix.raise "Extraneous arguments on the command line, got: #{inspect remaining_args}"
     end
 
     project = to_string(config[:name] || config[:app])
     version = config[:version] || "dev"
     options =
       config
       |> get_docs_opts()
-      |> Keyword.merge(cli_opts)
-      |> normalize_source_url(config)   # accepted at root level config
-      |> normalize_homepage_url(config) # accepted at root level config
+      # CLI options have presedence over config
+      |> Keyword.merge(cli_options)
+      # source_url is accepted at root level config
+      |> (&(if value = config[:source_url],
```

Comment:
At least for me, this *anonymous* function trick brakes the normal *workflow* of the data transformation. I like the way it was before, `normalize_source_url/1` and `normalize_homepage_url/1` are explicit and show their intention without looking the implementation details, if you want to look into the details you can go straight to that function. wdyt?